java/arcs/core/data/Ttl.kt

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.core.data
+
+/**  A class containing retention policy information. */
+data class Ttl(
+    val count: Int,
+    val units: Units?
```

Comment:
It could be good to assert that `(count >= 0 && units != null) || (count == -1 && units == null)` in an init block.
```kotlin
data class Ttl(
    val count: Int,
    val units: Units?
) {
    init {
        require((count >= 0 && units != null) || (count == -1 && units == null)) {
             "count must be non-negative when units are provided, or -1 if units is null"
        }
    }

    ...
}
```

Reply:
done, thanks!
wondering, is there an easy way to enforce that no other instances of `Infinite` are created? so that whoever wants to use `Infinite` uses the constant, or otherwise only creates a normal ttl?

Reply:
you could use a `sealed class`

e.g.

```kotlin
sealed class Ttl(open val count: Int, val isInfinite: Boolean = false) {
    val minutes: Int = count * when(this) {
        is Minutes -> 1
        is Hours -> 60
        ...
        is Infinity -> throw UnsupportedOperationException("...")
    }    

    data class Minutes(override val count: Int) : Ttl(count)
    data class Hours(override val count: Int) : Ttl(count)
    ...
    object Infinite : Ttl(-1, true)
}
```

Reply:
(Objects are singletons)